tests/unit/schema/test_column_schemas.py

```diff
@@ -296,3 +296,32 @@ def test_setting_flags_updates_shape_and_value_counts():
     assert updated_schema.properties["value_count"] == {"min": 0, "max": None}
     assert updated_schema.is_list
     assert updated_schema.is_ragged
+
+
+def test_pipe_merge():
+    col_schema_a = ColumnSchema(
+        "col_a", dtype=md.float32, properties={"value_count": {"min": 3, "max": 5}}
+    )
+    col_schema_b = ColumnSchema(
+        "col_b",
+        dtype=md.float64,
+        is_list=True,
+        is_ragged=False,
+        properties={"value_count": {"min": 5, "max": 5}},
+    )
+
+    new_schema = col_schema_b | col_schema_a
```

Comment:
I wonder if matching the behaviour of Python dict union would provide a more familiar experience for users of this operator?

> If a key appears in both operands, the last-seen value (i.e. that from the right-hand operand) wins
> https://peps.python.org/pep-0584/

Reply:
Oh interesting, that's a fair point. I suggested the other order because it matches the semantics of `or` where the first argument is returned unless it's false-y, otherwise the second.